spec/requests/text_extraction_spec.rb

```diff
@@ -5,50 +5,72 @@
-      expect(page.all('.selected-item-label').count).to eq 1
+        find('[data-text-extraction-label="Adyghe"]').click
 
-      find('[data-text-extraction-label="English"]').click
-      expect(page.all('.selected-item-label').count).to eq 2
+        expect(page).to have_css 'div', text: 'Selected language(s)'
+        expect(page.all('.selected-item-label').count).to eq 1
+
+        find('[data-text-extraction-label="English"]').click
+        expect(page.all('.selected-item-label').count).to eq 2
+      end
+    end
+
+    describe '#create' do
+      it 'adds ocrWF' do
+        post "/items/#{druid}/text_extraction", params: { text_extraction_languages: ['English'] }
+        expect(workflow_client).to have_received(:create_workflow_by_name).with(druid, 'ocrWF', context: { manuallyCorrectedOCR: false, ocrLanguages: ['English'] }, version: 2)
+        expect(object_client).to have_received(:reindex)
+        expect(response).to redirect_to(solr_document_path(druid))
+      end
     end
   end
 
-  describe '#create' do
-    let(:object_client) { instance_double(Dor::Services::Client::Object, reindex: true) }
-    let(:workflow_client) { instance_double(Dor::Workflow::Client, workflow:, create_workflow_by_name: nil, lifecycle: Time.zone.now) }
-    let(:workflow) { instance_double(Dor::Workflow::Response::Workflow, active_for?: false) }
-    let(:version_service) { instance_double(VersionService, open?: true) }
+  context 'when item is media' do
+    let(:object_type) { 'https://cocina.sul.stanford.edu/models/media' }
+
+    describe '#new' do
+      it 'shows text extraction form' do
+        visit "/items/#{druid}/text_extraction/new"
 
-    before do
-      allow(VersionService).to receive(:new).and_return(version_service)
-      allow(WorkflowClientFactory).to receive(:build).and_return(workflow_client)
-      allow(Dor::Services::Client).to receive(:object).and_return(object_client)
+        expect(page).to have_css 'h3', text: 'Text extraction'
+        expect(page).to have_content 'Avoid auto-generating caption/transcript files for media that do not contain any speech or lyrics'
+      end
     end
 
-    it 'adds ocrWF' do
-      post "/items/#{druid}/text_extraction", params: { text_extraction_languages: ['English'] }
-      expect(workflow_client).to have_received(:create_workflow_by_name).with(druid, 'ocrWF', context: { manuallyCorrectedOCR: false, ocrLanguages: ['English'] }, version: 2)
-      expect(object_client).to have_received(:reindex)
-      expect(response).to redirect_to(solr_document_path(druid))
+    describe '#create' do
+      it 'adds speechToTextWF' do
+        post "/items/#{druid}/text_extraction", params: {}
+        expect(workflow_client).to have_received(:create_workflow_by_name).with(druid, 'speechToTextWF', context: {}, version: 2)
+        expect(object_client).to have_received(:reindex)
+        expect(response).to redirect_to(solr_document_path(druid))
+      end
     end
   end
```

Comment:
the diff here is hard to read, but basically i just ensured that the correct workflow is started for media vs book/image/document